tokio-console/src/config.rs

```diff
@@ -394,33 +440,34 @@ impl ConfigFile {
         Ok(config)
     }
 
-    fn into_view_options(self) -> ViewOptions {
-        ViewOptions {
-            no_colors: self.no_colors(),
-            lang: self.charset.as_ref().and_then(|config| config.lang.clone()),
-            ascii_only: self.charset.as_ref().and_then(|config| config.ascii_only),
-            truecolor: self.colors.as_ref().and_then(|config| config.truecolor),
-            palette: self.colors.as_ref().and_then(|config| config.palette),
-            toggles: ColorToggles {
-                color_durations: self.color_durations(),
-                color_terminated: self.color_terminated(),
-            },
-        }
+    fn target_addr(&self) -> color_eyre::Result<Option<Uri>> {
+        let uri = self
+            .target_addr
+            .as_ref()
+            .map(|addr| addr.parse::<Uri>())
+            .transpose()
+            .wrap_err(format!("failed to parse Uri, {:?}", self.target_addr))?;
+        Ok(uri)
     }
 
-    fn from_view_options(view_options: ViewOptions) -> Self {
-        Self {
-            charset: Some(CharsetConfig {
-                lang: view_options.lang,
-                ascii_only: view_options.ascii_only,
-            }),
-            colors: Some(ColorsConfig {
-                enabled: view_options.no_colors.map(Not::not),
-                truecolor: view_options.truecolor,
-                palette: view_options.palette,
-                enable: Some(view_options.toggles),
-            }),
-        }
+    fn env_filter(&self) -> color_eyre::Result<Option<tracing_subscriber::EnvFilter>> {
+        let env_filter = self
+            .env_filter
+            .as_ref()
+            .map(|directive| directive.parse::<tracing_subscriber::EnvFilter>())
+            .transpose()
+            .wrap_err(format!("failed to parse EnvFilter, {:?}", self.env_filter))?;
+        Ok(env_filter)
+    }
+
+    fn retain_for(&self) -> color_eyre::Result<Option<RetainFor>> {
+        let retain_for = self
+            .retain_for
+            .as_ref()
+            .map(|retain_for| retain_for.parse::<RetainFor>())
+            .transpose()
+            .wrap_err(format!("failed to parse RetainFor, {:?}", self.retain_for))?;
```

Comment:
```suggestion
            .wrap_err_with(|| format!("failed to parse retention interval {:?} as a duration", self.retain_for))?;
```

Reply:
Resolved. Because I implemented `Deserialize` for `RetainFor`. So we don't need to parse the string it.